packages/pipeline-editor/src/pipeline-hooks.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2018-2021 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RequestHandler } from '@elyra/services';
+import {
+  pipelineIcon,
+  kubeflowIcon,
+  airflowIcon,
+  pyIcon,
+  rIcon
+} from '@elyra/ui-components';
+import { LabIcon, notebookIcon } from '@jupyterlab/ui-components';
+import produce from 'immer';
+import useSWR from 'swr';
+
+interface IReturn<T> {
+  data?: T | undefined;
+  error?: any;
+}
+
+interface IRuntimeImagesResponse {
+  'runtime-images': IRuntimeImage[];
+}
+
+interface IRuntimeImage {
+  name: string;
+  display_name: string;
+  metadata: {
+    image_name: string;
+  };
+}
+
+export const useRuntimeImages = <T = IRuntimeImagesResponse>(): IReturn<T> => {
```

Comment:
IIUC this is replacing `PipelineService.getRuntimes` but it removing the error handling in that function?

Reply:
oh yea, so I should reject if the array is empty? I also missed the sorting

Reply:
we should keep the same functionality, but based on my understanding of swr from this PR should we not just put the swr functionality in the MetadataService rather than bypassing it? then any calls to it would just be supported, or is it not that simple?